### PR TITLE
Adds PostCSS and some useful utilities.

### PR DESCRIPTION
**Pull request by NikolaRHristov**

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -1,12 +1,15 @@
 /** @type {import('tailwindcss').Config} */
 module.exports = {
-	content: ["./src/**/*.{astro,html,jsx,tsx,svelte,vue,js,ts}"],
+	content: [
+		"./public/**/*.html",
```
Comment (shaunchander) [inline tailwind.config.js]: I don't this this change is needed, by convention all of our views should live inside of `/src`. If someone needs to support `/public` then they should add this key for their use case.

File: postcss.config.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+		require("tailwindcss/nesting"),
+		require("tailwindcss")("./tailwind.config.js"),
+		require("postcss-combine-media-query"),
+		require("postcss-combine-duplicated-selectors")({
+			removeDuplicatedProperties: true,
+			removeDuplicatedValues: false,
+		}),
+		require("autoprefixer"),
+		require("cssnano")({ preset: "advanced" }),
+		require("postcss-reporter"),
```
Comment (shaunchander) [inline postcss.config.js]: Could you let me know how these PostCSS plugins affect the outputted CSS compared to vanilla TailwindCSS w/autoprefixer? Curious to know if `postcss-combine-media-query` et al. help in reducing/optimizing the CSS file size.

File: .github/workflows/node.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            ITERATIVE_DO_NOT_TRACK: 1
+            NEXT_TELEMETRY_DEBUG: 1
+            NEXT_TELEMETRY_DISABLED: 1
+            NG_CLI_ANALYTICS: false
+            NUXT_TELEMETRY_DISABLED: 1
+            PIN_DO_NOT_TRACK: 1
+            POWERSHELL_TELEMETRY_OPTOUT: 1
+            SAM_CLI_TELEMETRY: 0
+            STNOUPGRADE: 1
+            STRIPE_CLI_TELEMETRY_OPTOUT: 1
```
Comment (shaunchander) [inline .github/workflows/node.yml]: There seems to be a lot of default envs getting set here, I don't think we want some of these for an Astro starter.

File: .github/workflows/node.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            - run: pnpm install
+              working-directory: .
+
+            - run: pnpm run build
+              working-directory: .
+
+            - uses: actions/upload-artifact@v3.1.2
+              with:
+                  name: .-node-${{ matrix.node-version }}-dist
+                  path: ./dist
```
Comment (shaunchander) [inline .github/workflows/node.yml]: Not sure if we need to upload artifacts? (I'm assuming this CI just checks for passing builds)

File: .github/dependabot.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+version: 2
+updates:
+    - package-ecosystem: "github-actions"
+      directory: "/"
+      schedule:
+          interval: "daily"
```
Comment (shaunchander) [inline .github/dependabot.yml]: I'd opt for weekly package checks rather than daily. Less spam.

**Comment by NikolaRHristov** [comment]
My bad I seem to have pushed more to the repository than necessary I'll correct and resubmit again.

**Comment by shaunchander** [comment]
@nikolaxhristov Please also commit your updated `package-lock.json` file since new packages were added!

**Comment by NikolaRHristov** [comment]
Updated!

**Comment by NikolaRHristov** [comment]
Hi, @shaunchander I've updated the `pnpm-lock.yaml`

**Comment by shaunchander** [comment]
Thanks @nikolaxhristov, will be taking a look at this tomorrow!

**Comment by shaunchander** [comment]
@nikolaxhristov I'm a little bit hesitant on merging this PR since these CSS plugins seem to be project dependent? I personally don't use a lot of these plugins in my projects but I'm wondering if other developers would find value in having them shipped by default with the starter.

**Pull request closed without merging**